### PR TITLE
Customize images

### PR DESCRIPTION
**Pull request by dperpeet**
replaces #3274

### CODE REVIEW

**Comment by stefwalter** [comment]
So with this can we remove test/vm-install ?

**Comment by dperpeet** [comment]
> So with this can we remove test/vm-install ?

No. vm-install also builds packages (in the target system) and uses the image specific install script, which is for cockpit installation.

File: test/vm-customize [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    allscripts = []
+    for foo in scriptlist:
+        if os.path.isfile(foo):
+            pname = os.path.basename(foo)
+            uploadpath = "/var/tmp/" + pname
+            machine_instance.upload([foo], uploadpath)
+            machine_instance.execute("chmod a+x %s" % uploadpath)
+            machine_instance.execute(uploadpath)
+        else:
+            machine_instance.message("Bad path to script: " + foo)
```
Comment (stefwalter) [inline test/vm-customize]: This message should show up when not verbose. It's an error, right?

File: test/vm-customize [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    """
+    allpackages = []
+    for foo in packagelist:
+        if os.path.isfile(foo):
+            pname = os.path.basename(foo)
+            machine_instance.upload([foo], "/var/tmp/" + pname)
+            allpackages.append("/var/tmp/" + pname)
+        elif not re.search("/", foo):
+            allpackages.append(foo)
+        else:
```
Comment (stefwalter) [inline test/vm-customize]: This message should show up when not verbose. It's an error, right?

Comment (dperpeet) [inline test/vm-customize]: > This message should show up when not verbose. It's an error, right?

Yes. I changed it to use `sys.stderr`.

File: test/vm-customize (hunk 1 above)
Comment (dperpeet) [inline test/vm-customize]: > This message should show up when not verbose. It's an error, right?

Yes. I changed it to use `sys.stderr`.

**Comment by dperpeet** [comment]
@jscotka are you ok with the changes?

**Comment by jscotka** [comment]
Hi,
Yes I'm okay with these changes. 
Maybe as @stefwalter mentioned it would be nice to remove duplicate code from vm-install, and instead of that in vm-isntall import this  there and use.

**Comment by stefwalter** [comment]
Merged into git master.

**Pull request closed without merging**